File: JavaScript/JavaScriptProgrammingBasics/L02_ConditionalStatements/Exercises/P04_MetricConverter.js
function metricConverter(a, b, c) {
    let num = Number(a)
    let inputMeasure = b
    let outputMeasure = c

    if (inputMeasure === "mm" && outputMeasure === "m") {
        console.log((num / 1000).toFixed(3));
    } else if (inputMeasure === "mm" && outputMeasure === "cm") {
        console.log((num / 10).toFixed(3));

    } else if (inputMeasure === "m" && outputMeasure === "cm") {
        console.log((num * 100).toFixed(3));
    } else if (inputMeasure === "m" && outputMeasure === "mm") {
        console.log((num * 1000).toFixed(3));

    } else if (inputMeasure === "cm" && outputMeasure === "mm") {
        console.log((num * 10).toFixed(3));
    } else if (inputMeasure === "cm" && outputMeasure === "m") {
        console.log((num / 100).toFixed(3));
    }
}

File: C++/M01_ProgrammingBasics/L02_ConditionalStatements/Exercises/Solutions/P04_MetricConverter.cpp
#include <iostream>
using namespace std;

int main() {

	double num;
	string inputMeasure, outputMeasure;
	cin >> num >> inputMeasure >> outputMeasure;

	if (inputMeasure == "mm") {
		num /= 1000;
	} else if (inputMeasure == "cm") {
		num /= 100;
	}

	if (outputMeasure == "mm") {
		num *= 1000;
	} else if (outputMeasure == "cm") {
		num *= 100;
	}

	cout.setf(ios::fixed);
	cout.precision(3);

	cout << num << endl;

	return 0;
}
